src/main/scala/mesosphere/marathon/core/appinfo/TaskCounts.scala

```diff
@@ -13,16 +13,34 @@ import org.apache.mesos.{ Protos => mesos }
   * @param tasksUnhealthy snapshot of the number of unhealthy tasks (does not include tasks without health info)
   */
 case class TaskCounts(
-  tasksStaged: Int,
-  tasksRunning: Int,
-  tasksHealthy: Int,
-  tasksUnhealthy: Int)
+    tasksStaged: Int,
+    tasksRunning: Int,
+    tasksHealthy: Int,
+    tasksUnhealthy: Int) {
+  def +(counts: TaskCounts): TaskCounts = {
+    copy(
+      tasksRunning = tasksRunning + counts.tasksRunning,
+      tasksStaged = tasksStaged + counts.tasksStaged,
+      tasksHealthy = tasksHealthy + counts.tasksHealthy,
+      tasksUnhealthy = tasksUnhealthy + counts.tasksUnhealthy
+    )
+  }
+
+  def -(counts: TaskCounts): TaskCounts = {
+    copy(
+      tasksRunning = tasksRunning - counts.tasksRunning,
+      tasksStaged = tasksStaged - counts.tasksStaged,
+      tasksHealthy = tasksHealthy - counts.tasksHealthy,
+      tasksUnhealthy = tasksUnhealthy - counts.tasksUnhealthy
```

Comment:
Well, it wouldn't work like I use it. And I don't agree. Why should it? I actually think if you really want to restrict the counts to positive numbers, your example should result in an exception. But I think the chosen semantic is completely valid.

Semantically, you get a TaskCount diff -- I could introduce another type for that but I think that this is not necessary. I can't imagine someone using this method without thinking about the semantics. So the error potential is small.